src/automaton/levenshtein.rs

```diff
@@ -165,27 +163,57 @@ impl DynamicLevenshtein {
     }
 }
 
+/// Levenshtein automaton state.
+///
+/// It is useful for obtaining edit distance while searching.
+/// See examples in `Map::search_with_state` or `Set::search_with_state` documentation.
+///
+/// This is only defined when the `levenshtein` crate feature is enabled.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct LevenshteinState {
+    /// Internal state index.
+    pub state_idx: usize,
+    /// Levenshtein edit distance.
+    pub distance: Option<usize>,
+}
+
 impl Automaton for Levenshtein {
-    type State = Option<usize>;
+    type State = Option<LevenshteinState>;
```

Comment:
Sorry, but I'm not putting out a `0.5` release for this. :-(

Other ideas:

* Keep this PR on hold until I'm ready for another breaking change release.
* Create a new levenshtein automaton type.
* Put this automaton in a new crate.

Reply:
I would choose 1. option - on hold. 
I can inline updated automaton into my project - I don't need it in the official `fst` release. So merge it when you want and I think this PR can be used also as an example when other users want to use automaton states.

Reply:
Great. Thanks for understanding.